fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterContext.java

```diff
@@ -153,8 +154,11 @@ public void removeFilter(ExprId targetId, PhysicalHashJoin builderNode) {
         if (filters != null) {
             Iterator<RuntimeFilter> iter = filters.iterator();
             while (iter.hasNext()) {
-                if (iter.next().getBuilderNode().equals(builderNode)) {
+                RuntimeFilter rf = iter.next();
+                if (rf.getBuilderNode().equals(builderNode)) {
+                    builderNode.getRuntimeFilters().remove(rf);
                     iter.remove();
+                    prunedRF.add(rf);
```

Comment:
Seems that prunedRF is not used further?

Reply:
it is for ut. 
ut comes latter.